Classes/Dimension/ContentDimension.php

```diff
@@ -14,160 +10,121 @@
  * source code.
  */
 
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\DimensionSpace\Dimension;
+
+use Neos\ContentRepository\DimensionSpace\Dimension\Exception\ContentDimensionValuesAreMissing;
 use Neos\Utility\Arrays;
 
 /**
  * The content dimension domain model
  */
 final class ContentDimension
 {
-    /**
-     * @var ContentDimensionIdentifier
-     */
-    protected $identifier;
+    public readonly ContentDimensionIdentifier $identifier;
 
     /**
      * An array of dimension values, indexed by their identifiers
      *
-     * @var array|ContentDimensionValue[]
+     * @var array<string,ContentDimensionValue>
      */
-    protected $values = [];
+    public readonly array $values;
 
     /**
      * all Content Dimension Values indexed by "specialization", so
      * you can answer questions like "what's the next-most generic value for the given value"
      *
-     * @var array|ContentDimensionValue[]
+     * @var array<string,ContentDimensionValue>
      */
-    protected $generalizations;
+    public readonly array $generalizations;
 
     /**
      * all Content Dimension Values indexed by "generalization", so
      * you can answer questions like "what are the next-most specialized values for the given value"
      *
      * returns an *array* of specializations for each key; so this is effectively an array of arrays
      *
-     * @var array|ContentDimensionValue[][]
+     * @var array<string,array<string,ContentDimensionValue>>
      */
-    protected $specializations;
+    public readonly array $specializations;
 
-    /**
-     * @var ContentDimensionValue
-     */
-    protected $defaultValue;
+    public readonly ContentDimensionValue $defaultValue;
 
     /**
      * General configuration like UI, detection etc.
-     * @var array
+     * @var array<string,mixed>
      */
-    protected $configuration;
+    public readonly array $configuration;
 
-    /**
-     * @var ContentDimensionValueSpecializationDepth
-     */
-    protected $maximumDepth;
+    public readonly ContentDimensionValueSpecializationDepth $maximumDepth;
 
     /**
-     * @param ContentDimensionIdentifier $identifier
-     * @param array|ContentDimensionValue[] $values
-     * @param ContentDimensionValue $defaultValue
-     * @param array|ContentDimensionValueVariationEdge[] $variationEdges
-     * @param array $configuration
-     * @throws Exception\ContentDimensionValuesAreMissing
+     * @param array<string,ContentDimensionValue> $values
+     * @param array<string,mixed> $configuration
+     * @throws ContentDimensionValuesAreMissing
```

Comment:
Incomplete :)

Reply:
I prefer to only use parameter annotations where they add information. The arrays here help tools like PHPStan with static analysis, while we don't win anything by copy&pasting properly typed parameters to the comments. Is there any PSR for this?